src/nmod_vec.h

```diff
@@ -78,121 +78,104 @@ void _nmod_vec_swap(nn_ptr a, nn_ptr b, slong length)
-                    }                                                       \
-                }                                                           \
-                else                                                        \
-                {                                                           \
-                    ulong v0, v1, u0, u1;                               \
-                    i = 0;                                                  \
-                    if ((len) & 1)                                          \
-                        umul_ppmm(v1, v0, (expr1), (expr2));                \
-                    else                                                    \
-                        v0 = v1 = 0;                                        \
-                    for (i = (len) & 1; i < (len); i++)                     \
-                    {                                                       \
-                        umul_ppmm(t1, t0, (expr1), (expr2));                \
-                        add_ssaaaa(s1, s0, s1, s0, t1, t0);                 \
-                        i++;                                                \
-                        umul_ppmm(u1, u0, (expr1), (expr2));                \
-                        add_ssaaaa(v1, v0, v1, v0, u1, u0);                 \
-                    }                                                       \
-                    add_ssaaaa(s1, s0, s1, s0, v1, v0);                     \
-                }                                                           \
-                NMOD2_RED2(s0, s1, s0, mod);                                \
-                break;                                                      \
-            default:                                                        \
-                for (i = 0; i < (len); i++)                                 \
-                {                                                           \
-                    umul_ppmm(t1, t0, (expr1), (expr2));                    \
-                    add_sssaaaaaa(s2, s1, s0, s2, s1, s0, 0, t1, t0);       \
-                }                                                           \
-                NMOD_RED(s2, s2, mod);                                      \
-                NMOD_RED3(s0, s2, s1, s0, mod);                             \
-                break;                                                      \
-        }                                                                   \
-        res = s0;                                                           \
-    } while (0);
-
-ulong _nmod_vec_dot(nn_srcptr vec1, nn_srcptr vec2,
-    slong len, nmod_t mod, int nlimbs);
-
-ulong _nmod_vec_dot_rev(nn_srcptr vec1, nn_srcptr vec2,
-    slong len, nmod_t mod, int nlimbs);
-
-ulong _nmod_vec_dot_ptr(nn_srcptr vec1, const nn_ptr * vec2, slong offset,
-    slong len, nmod_t mod, int nlimbs);
+#define NMOD_VEC_DOT(res, i, len, expr1, expr2, mod, nlimbs) \
```

Comment:
Why the nonstandard level of indentation?

Reply:
More compact format in headers. Can revert this.